tests/bsim/bluetooth/host/misc/sample_test/test_scripts/run.sh

```diff
@@ -0,0 +1,80 @@
+#!/usr/bin/env bash
+# Copyright (c) 2024 Nordic Semiconductor
+# SPDX-License-Identifier: Apache-2.0
+
+set -eu
+
+# Provides common functions for bsim tests.
+# Mainly `Execute`, and `wait_for_background_jobs`.
+source ${ZEPHYR_BASE}/tests/bsim/sh_common.source
+
+# Helper variable. Expands to "tests_bsim_bluetooth_host_misc_sample_test".
+test_name="$(guess_test_long_name)"
+
+# Use a unique simulation id per test script. It is a necessity as the CI runner
+# will run all the test scripts in parallel. If multiple simulations share the
+# same ID, they will step on each other's toes in unpredictable ways.
+simulation_id=${test_name}
+
+# This controls the verbosity of the simulator. It goes up to 9 (not 11, sorry)
+# and is useful for figuring out what is happening on the PHYsical layer (e.g.
+# device 1 starts listening at T=100us) among other things.
+# `2` is the default value if not specified.
+verbosity_level=2
+
+# This sets the watchdog timeout for the `Execute` function.
+#
+# Although all simulations are started with a bounded end (the `sim_length`
+# option), something very wrong can still happen and this additional time-out
+# will ensure all executables started by the current script are killed.
+#
+# It measures wall-clock time, not simulated time. E.g. a test that simulates 5
+# hours might actually complete (have a runtime of) 10 seconds.
+#
+# The default is set in `sh_common.source`.
+# Guidelines to set this value:
+# - Do not set it to a value lower or equal to the default.
+# - If the test takes over 5 seconds of runtime, set `EXECUTE_TIMEOUT` to at
+#   least 5 times the run-time on your machine.
+EXECUTE_TIMEOUT=120
+
+# Set simulation length, in microseconds. The PHY will run for this amount of
+# simulated time, unless both devices exit.
+#
+# If you are not early-exiting the devices (e.g. using `TEST_PASS_AND_EXIT()`),
+# please run the test once and set the simulation time in the same ballpark. No
+# need to simulate hours of runtime if the test finishes in 10 seconds.
+#
+SIM_LEN_US=$((2 * 1000 * 1000))
```

Comment:
One quite relevant thing about this, is that in general setting this -test_deadline to the same value as -sim_length will not result in the test tick callback being called.
The reason being that the phy kills the devices when that sim_length is just happening, not after.
So if the devices were in the middle of a Tx or Rx, they will be killed by the phy before they get to do the test callback.
You'd need to set this to be a few µs less in devices to ensure the callback is called.
(you can see it happen by setting this value to 969860 for ex.)

Reply:
I see. 
I had it a few us before initially but it was always called in my local runs.
I think it makes more sense to do the adjustment in `timeout.c` as it would be extra boilerplate "polluting" every test script.